### DISCUSSION TITLE
Adaptive layout with StateContainer

### DISCUSSION THREAD

**Discussion by Strypper** [CommunityToolkit/Maui#860] [New Feature Discussions]
Since the toolkit introduced the StateContainer I see this is a huge potential for cross-platform solution. The way we pick the right layout shouldn’t rely on the device info but should rely the current width the app window because the Galaxy Z Fold can folded and unfolded and Ipad can open application in narrow state for splitscreen and yes this work with Android and IPad OS also. the quickest way would be create the whole ui for 3 states phone - tablet - desktop. Sadly the StateContainer keep crashing when I change the state too fast with the sizechanged event, and it force me to use the grid or the stack to host the stateview —> this will also kill the scrolling if I have the collectionview in it, hope the team make this work for the content page itself

**Comment by pictos** [comment]
You can do something like this with [ContentTemplate](https://learn.microsoft.com/en-us/xamarin/xamarin-forms/app-fundamentals/templates/control-template) and probably can use a `DataTemplateSelector`. Not sure if we can add more value on that, but feel free to think in an API design for this feature

**Reply by Strypper** [reply]
> You can do something like this with [ContentTemplate](https://learn.microsoft.com/en-us/xamarin/xamarin-forms/app-fundamentals/templates/control-template) and probably can use a `DataTemplateSelector`. Not sure if we can add more value on that, but feel free to think in an API design for this feature

If iam not wrong you can only use datatemplateselector when you using listview or collectionview 

But with my case, I might using the collectionview but the things I wanna change for bigger screens would be the ItemLayout (I still haven't find a way to change it programmatically) which have nothing to do with templateselecotr

**Reply by pictos** [reply]
@Strypper not sure, I could use it with BindableLayout... It's something to try out. Please, share with us your finds (:

**Comment by CliffAgius** [comment]
Closed and answered
